### PR TITLE
Update source tree for zig 0.8.0

### PR DESCRIPTION
**Pull request by kjaleshire**
Updates certain backwards-incompatible std calls, including:

* [Rename](https://ziglang.org/download/0.8.0/release-notes.html#importbuiltin-no-longer-re-exports-stdbuiltin) `@import("builtin");` to `std.builtin`
* [Rename](https://ziglang.org/download/0.8.0/release-notes.html#importbuiltin-no-longer-re-exports-stdbuiltin) (scroll down) `builtin.arch` to `builtin.cpu.arch` 
* One instance of changing the `assert` surrounding `AutoHashMap.remove` calls.

And one language-specific change:
* Replace `callconv(.Inline)` with keyword `inline`

### CODE REVIEW

**Review by DrDeano — CHANGES_REQUESTED** [review]
The format check failed, please check by running `zig build` to run the format.

**Comment by kjaleshire** [comment]
The `zig fmt` call is failing on `callconv(.Inline)` => `inline`. The error is erroneous and will disappear once the build checks use the release version of 0.8.0.

**Comment by SamTebbs33** [comment]
Thanks for doing this @kjaleshire ! Sorry for not looking at it earlier, I've had an issue with GitHub not sending me notifications for everything I want it to. I've added a change that will make the CI use 0.8.0 so that the checks pass.

**Comment by DrDeano** [comment]
Ah, yes, I forgot we locked the zig version for the CI, sorry about that.

**Comment by SamTebbs33** [comment]
I've added `try` before each `expect` function call but 0.8.0 has the issue that made us lock the version in the first place so we won't be able to merge this yet, unfortunately. This could become our zig update PR once the issue is fixed in master.

**Comment by SamTebbs33** [comment]
After a long time away from kernel dev, I've managed to reduce the blockers for this PR to a seg fault when running the kernel. At least the tests are passing now!

**Comment by SamTebbs33** [comment]
I submitted PR 9939 to fix the zig segfault. Once that is merged we can go back to master.

File: README.md
Comment (DrDeano) [inline README.md]: This is now 0.9.1

File: build.zig
Comment (DrDeano) [inline build.zig:118]: How come this has changed to enable qemu for the entire build? As this was used when running the unit tests only.

File: src/kernel/arch/x86/gdt.zig
Comment (DrDeano) [inline src/kernel/arch/x86/gdt.zig:9]: Given that the `mock_path` is not being used, could remove it from `build.zig` as well.

File: src/kernel/arch/x86/gdt.zig
Comment (DrDeano) [inline src/kernel/arch/x86/gdt.zig]: Given tests are run in all test modes, should catch panic rather than catch unreachable.

File: src/kernel/arch/x86/idt.zig
Comment (DrDeano) [inline src/kernel/arch/x86/idt.zig]: Same here.

File: src/kernel/arch/x86/idt.zig
Comment (DrDeano) [inline src/kernel/arch/x86/idt.zig]: I assume this debug print can be removed

File: src/kernel/arch/x86/keyboard.zig
Comment (DrDeano) [inline src/kernel/arch/x86/keyboard.zig]: What is your opinion that this could be a `warn`? As we would be losing key presses, so isn't really info, and is more of a warning.

File: src/kernel/arch/x86/pit.zig
Comment (DrDeano) [inline src/kernel/arch/x86/pit.zig]: How come these are commented out?

File: src/kernel/arch/x86/pit.zig
Comment (DrDeano) [inline src/kernel/arch/x86/pit.zig]: Same here, commented code?

File: test/mock/kernel/mock_framework_template.zig
Comment (DrDeano) [inline test/mock/kernel/mock_framework_template.zig]: No catch unreachable, should panic because of release-fast/small.
Same for the unreachable's below

File: test/mock/kernel/vga_mock.zig
Comment (DrDeano) [inline test/mock/kernel/vga_mock.zig]: Panic, over unreachable

File: test/runtime_test.zig
Comment (DrDeano) [inline test/runtime_test.zig:218]: Finally, the normal way to wait on a thread with `join` lol

File: src/kernel/arch/x86/multiboot.zig
Comment (DrDeano) [inline src/kernel/arch/x86/multiboot.zig:1]: I assume these changes are from the translate c of the multiboot.c

File: src/kernel/arch/x86/paging.zig
Comment (DrDeano) [inline src/kernel/arch/x86/paging.zig:687]: I assume this is needed because the compiler optimises the previous `var value = ptr.*;` out. If so, might be worth adding a comment as having prints in a test is odd.

File: src/kernel/filesystem/vfs.zig
Comment (DrDeano) [inline src/kernel/filesystem/vfs.zig]: Catch panic

File: src/kernel/heap.zig
Comment (DrDeano) [inline src/kernel/heap.zig]: Why not use the `Self` defined?

File: src/kernel/filesystem/fat32.zig
Comment (DrDeano) [inline src/kernel/filesystem/fat32.zig]: Could this test be skipped instead of removed, as in the future, this could be fixed.

File: src/kernel/kmain.zig
Comment (DrDeano) [inline src/kernel/kmain.zig]: Given this is now move lower down, there are still calls to panic above. If these `panic` calls are called, what will happen? Will it just print address and no symbols. Might be worth changing the `init` function to `init_symbols` or something similar to indicate that this doesn't initialise the panic handler, but initialises the symbol mappings.

**Review by DrDeano — COMMENTED** [review]
I congratulate the effort for this :) I expect only the finest vim macros were used ;)

File: build.zig
Comment (SamTebbs33) [reply build.zig:118]: The `enable_qemu` field was been moved from there to the `Builder` unfortunately ([this commit](https://github.com/ziglang/zig/commit/b24cbecdb2abb4399d65553ebade318263cd57d3) for reference).

File: src/kernel/arch/x86/gdt.zig
Comment (SamTebbs33) [reply src/kernel/arch/x86/gdt.zig]: The program will exit if the `expectEqual` fails though, right? I don't think the `catch` part will ever run.

File: src/kernel/arch/x86/idt.zig
Comment (SamTebbs33) [reply src/kernel/arch/x86/idt.zig]: Yep, good spot.

File: src/kernel/arch/x86/keyboard.zig
Comment (SamTebbs33) [reply src/kernel/arch/x86/keyboard.zig]: Yeah that sounds better to me

File: src/kernel/arch/x86/pit.zig
Comment (SamTebbs33) [reply src/kernel/arch/x86/pit.zig]: Hmm I don't remember, I'll uncomment them and hope they pass...

Comment (SamTebbs33) [reply src/kernel/arch/x86/pit.zig]: They're commented out in develop too. I don't think I have the knowledge of the PIT to get them working again.

File: src/kernel/arch/x86/multiboot.zig
Comment (SamTebbs33) [reply src/kernel/arch/x86/multiboot.zig:1]: Indeedy

File: src/kernel/arch/x86/paging.zig
Comment (SamTebbs33) [reply src/kernel/arch/x86/paging.zig:687]: Yeah exactly. Good idea

File: src/kernel/heap.zig
Comment (SamTebbs33) [reply src/kernel/heap.zig]: That is actually within the `Header` struct, but I can add one for `FreeListAllocator` too.

File: src/kernel/filesystem/fat32.zig
Comment (SamTebbs33) [reply src/kernel/filesystem/fat32.zig]: Agreed. How do I do that? Just comment it out or is there a fancy way of skipping tests?

File: src/kernel/kmain.zig
Comment (SamTebbs33) [reply src/kernel/kmain.zig]: Yeah if something above this panics then it will just print the addresses without symbols. I like that idea, it makes the purpose clear.

File: build.zig
Comment (DrDeano) [reply build.zig:118]: Fair enough

File: src/kernel/filesystem/fat32.zig
Comment (DrDeano) [reply src/kernel/filesystem/fat32.zig]: Can return `error.SkipZigTest`. This can be found in `special\test_runner.zig`

File: src/kernel/arch/x86/gdt.zig
Comment (DrDeano) [reply src/kernel/arch/x86/gdt.zig]: `expectEqual` will return `error.TestExpectedEqual` if the values are not equal, so for ReleaseFast, the `catch unreachable` will be ignored and `expectEqual` won't fail the test

File: src/kernel/arch/x86/pit.zig
Comment (DrDeano) [reply src/kernel/arch/x86/pit.zig]: Interesting? I shall investigate

File: src/kernel/arch/x86/gdt.zig
Comment (SamTebbs33) [reply src/kernel/arch/x86/gdt.zig]: Yep you're right. I think it used to work in the way I was thinking.

File: src/kernel/filesystem/fat32.zig
Comment (SamTebbs33) [reply src/kernel/filesystem/fat32.zig]: NIce

File: src/kernel/arch/x86/pit.zig
Comment (DrDeano) [reply src/kernel/arch/x86/pit.zig]: I had a look and these were commented from the beginning. These commented lines can be removed as they don't relate to the test. I think they might be from a copy and paste error and forgot to remove them.

File: src/kernel/arch/x86/pit.zig
Comment (DrDeano) [reply src/kernel/arch/x86/pit.zig]: Can remove these lines

**Review by DrDeano — APPROVED** [review]

**Comment by SamTebbs33** [comment]
Thanks for the approval :) Looks like the checks are messing up. [This](https://stackoverflow.com/a/69169858) or temporarily allowing maintainers to bypass the checks may help.

**Comment by DrDeano** [comment]
> Thanks for the approval :) Looks like the checks are messing up. [This](https://stackoverflow.com/a/69169858) or temporarily allowing maintainers to bypass the checks may help.

Sure, can just temporary disable the check, then re-enable after the merge

**Comment by SamTebbs33** [comment]
Hmm it's still not working... I'll try submitting a new PR to see if it's just a problem with this one.

**Pull request merged**
